### PR TITLE
Categorize entity packages a bit

### PR DESCRIPTION
**Pull request by i509VCB**
The entity packages are quite big imo. So I've categorized the entities by their type.

For an entity to get it's own package, it must meet the criteria of having multiple variants (such as skeletons), or be distinctly abstract (such as raiders and illagers).

This also moves `Raid` and `RaidManager` to it's own package since these are completely seperate constructs from just an entity

### CODE REVIEW

File: mappings/net/minecraft/raid/Raid.mapping [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_3765 net/minecraft/entity/raid/Raid
```
Comment (liach) [inline mappings/net/minecraft/raid/Raid.mapping]: Should raid really go to root package... it is such a small thing

**Review by liach — CHANGES_REQUESTED** [review]
I personally disagree with splitting out fish, skeleton, and illager package. They can just go to their parent package.

File: mappings/net/minecraft/raid/Raid.mapping (hunk 1 above)
Comment (i509VCB) [reply mappings/net/minecraft/raid/Raid.mapping]: maybe `world` or `server.world`

Comment (liach) [reply mappings/net/minecraft/raid/Raid.mapping]: maybe `server.raid` package? or village (no raid subpackage)?

File: mappings/net/minecraft/entity/golem/IronGolemEntity.mapping [hunk 2]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_1439 net/minecraft/entity/passive/IronGolemEntity
+CLASS net/minecraft/class_1439 net/minecraft/entity/golem/IronGolemEntity
```
Comment (liach) [inline mappings/net/minecraft/entity/golem/IronGolemEntity.mapping]: Let's just move all golem to the entity root package, no golem subpackage

File: mappings/net/minecraft/raid/Raid.mapping (hunk 1 above)
Comment (i509VCB) [reply mappings/net/minecraft/raid/Raid.mapping]: Well villages are always the subject of a raid, so that makes sense

**Comment by i509VCB** [comment]
Personally I'd keep the skeleton and illager packages. Skeletons because they are distinctly skeletons.

Illagers because they are an official category of mobs

**Comment by liach** [comment]
Don't think grouping package by "official category" is a good idea. Java runtime library doesn't group collections into packages by what classes they extend.

**Comment by i509VCB** [comment]
We aren't the java runtime library though

**Comment by liach** [comment]
We don't have a reason to create super deep package trees. Super deep package trees with tiny packages are true nightmares

**Comment by i509VCB** [comment]
How small is too small and how deep is too deep?

**Comment by liach** [comment]
I think the current setup of packages is fairly fine. We just need to maybe split mob package into monster and neutral

**Comment by i509VCB** [comment]
Yeah I've noticed that, since mob is meant to be generic to any entity with the capabilities to do basic pathfinding and move itself

Probably a later PR tbf

File: mappings/net/minecraft/entity/mob/hoglin/HoglinBrain.mapping [hunk 3]
```diff
@@ -1,4 +1,4 @@
-CLASS net/minecraft/class_4835 net/minecraft/entity/mob/HoglinBrain
+CLASS net/minecraft/class_4835 net/minecraft/entity/mob/hoglin/HoglinBrain
```
Comment (YanisBft) [inline mappings/net/minecraft/entity/mob/hoglin/HoglinBrain.mapping:1]: Isn't there a package like `entity.ai.brain` or so for classes like these?

Comment (liach) [reply mappings/net/minecraft/entity/mob/hoglin/HoglinBrain.mapping:1]: it's where hoglin brain handling logic is, must be in same package as hoglin

**Comment by modmuss50** [comment]
I think we should put this on hold along with other large scale refactors untill 1.17 snapshots start (hopefully only a few months at max).

**Review by modmuss50 — APPROVED** [review]
As this is only package renames it should be fine to go in now.

**Comment by liach** [comment]
Per our discord discussion, this change will be on hold until 1.17 snapshots.

**Comment by i509VCB** [comment]
Did you scroll down far enough?

**Comment by liach** [comment]
Sure, but I still oppose this structural change. You split out way too many packages with too few classes.

**Comment by i509VCB** [comment]
This will probably require another PR, will reopen in 1.17 snapshots

**Pull request closed without merging**